internal/rdc/client.go

```diff
@@ -69,11 +75,12 @@ type readJobResponse struct {
 // New creates a new client.
 func New(url, username, accessKey string, timeout time.Duration, artifactConfig config.ArtifactDownload) Client {
 	return Client{
-		HTTPClient:     &http.Client{Timeout: timeout},
-		URL:            url,
-		Username:       username,
-		AccessKey:      accessKey,
-		ArtifactConfig: artifactConfig,
+		HTTPClient:          &http.Client{Timeout: timeout},
+		RetryableHTTPClient: retryablehttp.NewClient(),
```

Comment:
I don't think there's any harm in just using that client for all requests here. And just drop the regular client?

PS: I don't see a timeout defined for this client 🤔